### PR TITLE
Extend ECMA-262 syntax into a superset of JSON

### PR DESCRIPTION
**Pull request by gibson042**
This implements https://github.com/tc39/proposal-json-superset , currently at stage 3.

### CODE REVIEW

**Comment by ljharb** [comment]
Tests: https://github.com/tc39/test262/pull/1543

**Comment by littledan** [comment]
Thanks for your contribution, @gibson042 .

File: spec.html [hunk 1]
```diff
@@ -9908,7 +9908,7 @@ <h2>Syntax</h2>
...
+    <p>Like white space code points, line terminator code points are used to improve source text readability and to separate tokens (indivisible lexical units) from each other. However, unlike white space code points, line terminators have some influence over the behaviour of the syntactic grammar. In general, line terminators may occur between any two tokens, but there are a few places where they are forbidden by the syntactic grammar. Line terminators also affect the process of automatic semicolon insertion (<emu-xref href="#sec-automatic-semicolon-insertion"></emu-xref>). A line terminator cannot occur within any token except a |StringLiteral|, |Template|, or |TemplateSubstitutionTail|. &lt;LF&gt; and &lt;CR&gt; line terminators cannot occur within a |StringLiteral| token except as part of a |LineContinuation|.</p>
```
Comment (ljharb) [inline spec.html]: For clarity and future explorers, can you elaborate on why this restriction is narrowed?

Comment (gibson042) [reply spec.html]: It's the fundamental change of [proposal-json-superset](https://github.com/tc39/proposal-json-superset), just in a part of the spec that I didn't realize included affected text until working on https://github.com/tc39/test262/pull/1565 . As of this PR landing, _StringLiteral_ will be able to include any _LineTerminator_ as part of a _LineContinuation_ (backslash followed by _LineTerminatorSequence_, used to statically define long strings over multiple lines without affecting their content), and—invalidating the old text—will **also** be able to include U+2028 LINE SEPARATOR and U+2029 PARAGRAPH SEPARATOR line terminators in any position (as is the case for JSON strings).

Comment (mathiasbynens) [reply spec.html]: Previously, no line terminators could occur in string literals (except as part of a `LineContinuation`).

With the change, U+2028 and U+2029 can occur in string literals, even outside of `LineContinuation`.

In other words, this change clarifies that `LineContinuation`s are unaffected by the change, i.e. `\` followed by U+2028 or U+2029 in a string literal still evaluates to the empty string. This was already clear from the proposal, but updating this note as part of this patch improves accuracy.

**Comment by littledan** [comment]
This proposal [reached Stage 4](https://github.com/tc39/tc39-notes/blob/master/es9/2018-05/may-22.md#11ie-ecmascript-as-a-superset-of-json) in the May 2018 TC39 meeting. Is the patch ready to land?

**Comment by mathiasbynens** [comment]
Side note: this commit has just been merged, yet it doesn’t show up in the first few entries on https://github.com/tc39/ecma262/commits/master. Instead, it’s further down the list here:



Note that May 7th is the date the original PR was created.

Is this confusing to anyone else or just me? 

AFAICT until now, the commit log always followed merge order. Are we changing that now?

**Comment by ljharb** [comment]
This is actually a bug in github; it sorts by (and has sorted by) commit date, not actual merge order. cc @keithamus; is this something that we could request?

(The reason this happened here is that when I rebased the PR, i used `--committer-date-is-author-date`, which avoids obscuring the original commit date)

**Comment by mathiasbynens** [comment]
> (The reason this happened here is that when I rebased the PR, i used `--committer-date-is-author-date`, which avoids obscuring the original commit date)

This way, the commit order doesn’t make sense even in `git log`. Can we not do that please?

**Comment by ljharb** [comment]
My `git log` locally works in proper merge order (ie, starting at the branch HEAD, and moving backwards one commit parent at a time). I'm not sure why it's working differently for you; perhaps aliases, or an outdated git version?

**Comment by mathiasbynens** [comment]
You’re right, `git log` does show in proper merge order. My bad — I was confused again by the fact that a) the commit dates don’t match the merge dates for recent commits, as explained before, and b) [the commit overview](https://github.com/tc39/ecma262/commits/master) doesn’t link to the PRs for any of the recently-merged commits (another difference from how things have been done until now).

Please don’t use `--committer-date-is-author-date`, and include the PR links in the commit message when merging, just like we’ve always done for this repository.

**Comment by ljharb** [comment]
I'll definitely include the PR links; that's a convention that wasn't stressed in our editors' meeting (however, if you click on any commit, the header on github shows the PR numbers, branches, and tags that it's contained in - so that link isn't strictly required).

**Pull request merged**